components/code-codes/Claim/Claim.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState, FormEvent } from 'react';
+import toast from 'react-hot-toast';
+
+import OneInputForm from '@components/_ui/OneInputForm';
+import { CodecodesClaimPayload } from '@lib/types/codecodes';
+
+type Props = {
+  firstName?: string;
+  fullName?: string;
+  email?: string;
+};
+
+async function claim({ code, name, email }: CodecodesClaimPayload) {
+  const response = await fetch('/api/codecodes/claim', {
+    method: 'POST',
+    body: JSON.stringify({
+      code,
+      name,
+      email
+    })
```

Comment:
Não precisa desestruturar o objeto, acho

```suggestion
async function claim(payload: CodecodesClaimPayload) {
  const response = await fetch('/api/codecodes/claim', {
    method: 'POST',
    body: JSON.stringify(payload)
```

Reply:
Eu vou fazer outro PR pra ajustar esses esquemas do Code-Codes pq não tá funcionando nem o Claim e nem o Ranking, vou até tirar essas paradas dessa PR